back/engines/commercial/insights/app/models/insights/zeroshot_classification_task.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Insights
+  class ZeroshotClassificationTask < ::ApplicationRecord
+    has_and_belongs_to_many :categories, join_table: 'insights_zeroshot_classification_tasks_categories', foreign_key: :task_id
```

Comment:
Shouldn't we validate the presence of categories and inputs here ?

Reply:
Yeah, I hesitated a bit here. It's also a bit related to your next comment. Is it a good thing to have a task that has no categories or no inputs associated? 

Overall, I'm afraid that this kind of validation & callbacks could trigger quite a few extra DB queries that do not bring a lot. (I guess that's also why there is no native `presence` option for `has_many` association.) It was a bit different with the auto-tagging bc we were showing the number of ongoing tasks, but I don't think it's the case anymore.

I think it's fine in this case because, in the end, that's the payload of the message that will matter. And if we receive a message and the categories or inputs are not there anymore, it will silently die.

Reply:
What do you think?

Reply:
I didn't think of the relative db load, so indeed not always needed, but since these tasks relate to quite costly computations, cancelling any unused task could save quite some money and energy is important. That said, when creating the task, that shouldn't be an issue, if the nlp and BE dbs are synchronized right (which is still an ongoing concern atm, something that needs o be checked and improved), nlp shouldn't create useless tasks. But on category removal, it might still be needed.

Reply:
That could definitely save some energy 🌱 There is clearly something to be done about tasks cancellation. I propose as a first to step to cancel all tasks when all categories are cleared (low-hanging fruit) and implement 'cherrypicked' cancellation as a performance improvement later on. Would that work for u?

Reply:
Sounds good! :+1: